### ISSUE TITLE
static_filename - directories, lifetime

### ISSUE THREAD

**Issue by encukou** [pyvec/arca#58]
Two questions about [`static_filename`](https://arca.readthedocs.io/en/latest/cookbook.html#i-need-some-files-from-the-repositories) that aren't clear from the docs:

- Is it supposed to work on directories?
- For how long can I use the result reliably? (When can the file be garbage-collected or overwritten?)

**Comment by mikicz** [comment]
> It currently mostly works. But is this something I can rely on? Should I write a test for `static_filename(..., '.')`? Can I rely on `static_filename(...).parent` containing a file's siblings?

Yes, I think you should be able to rely on that. Arca's not messing around with the structure of the repository, so yes, `static_filename(...).parent` should contain the siblings.

> Right now I'm not sure what the lifetime is intended to be. Until the next call to Arca?

When `single_pull` is enabled than the lifetime is until a new pull (triggered by either a new init of Arca or by calling the method), otherwise the lifetime is until next call of `static_filename` or `run` (since either can trigger a new pull -> files can change)
